.github/workflows/security.yml

```diff
@@ -0,0 +1,65 @@
+name: Security-check
+
+on:
+  pull_request: # change to push before merging code
+    branches:
+      - main
+    paths:
+      - "**.py"
+      - ".github/workflows/*.yml"
+
+jobs:
+  Security-check:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
+        python-version: [3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Upgrade pip
+        run: |
+          pip install --upgrade --user pip
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-py${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ matrix.python-version }}-
+      - name: Install dependencies
+        run: |
+          hash -r
+          pip install -r requirements.dev.txt
+          pip install -e .
+
+      - id: file_changes  # get changed files.
+        uses: trilom/file-changes-action@v1.2.3
+        with:
+          output: ' '
+
+      - uses: pre-commit/action@v2.0.0
+        with:
+          extra_args: --files ${{ steps.file_changes.outputs.files}}  # apply only to changed files.
+
+      - name: Scan for security issues
+        run: |
+          bandit -r src
```

Comment:
Q: Why we have those here and also in tests? (should we remove them from tests?)

Reply:
Yes, this can be removed from tests. I have added them in both places because I was not sure where should they be placed. In an ideal situation, I feel all the 3, `bandit, safety and snyk` should run as `pr-tests`, similar to `python-tests` in this repo, but this cannot be done for `synk` because it will use a secret token and github doesn't provide these tokens to forked repos. **So we need to run snyk tests after we merge code into our codebase** (similar to benchmarking which uses github-token)